### ISSUE TITLE
Command times out when a process is (re-)started in the background

### ISSUE THREAD

**Issue by arminus** [OliveTin/OliveTin#152]
**What seems to be the problem?!**

I have a script which kills a running process, then restarts it in the background (i.e. with an & appended). This script is defined like so for OliveTin:

```
- title: Restart Overpass
  icon: restart
  shell: 'sudo -u www /home/www/osm/scripts/restart_overpass.sh'
  timeout: 60
```
The start command in the script is this:

```
${BASE}/overpass/bin/dispatcher --osm-base --db-dir=${BASE}/overpass/db &
sleep 5
dispID="`ps -ef|grep dispatcher|grep -v grep|awk '{print $2}'`"
echo "New dispatcher ps id: ${dispID}"
```

Apparently, OliveTin actually starts that process (with a parent ps ID of 1), but the exit code is OK (timed out) (i.e. the run button never restores to normal unless I reload the page). The command also clearly finishes within the 60 secs timeout.

Also tested this with a nohup, makes no difference.

**Can you provide a sosreport?**

Truncated in journalctl -eu OliveTin

**What package/file/container did you use to install OliveTin?**

.deb, Debian GNU/Linux 11 (bullseye), 2023.03.25

**Your config.yaml**

see above

**OliveTin logs**

![2023-08-04_124033](https://github.com/OliveTin/OliveTin/assets/683680/ed455f89-3b61-4bdd-88ab-22ef869abf2a)

**Screenshot of WebDeveloper console logs**

N/A

**Anything else?**

N/A

**Comment by jamesread** [comment]
Thanks for reporting this @arminus , I'll take a look and try and reproduce. Got a few other code things I need to prioritize for now, but will get to this when time allows.

**Comment by jamesread** [comment]
Hey @arminus , so sorry that this has taken such a long time for me to come and have a look at it. 

While investigating your issue, I found an error in the executor to do with killed processes not being handled properly, and have fixed that in #257 .

**Comment by jamesread** [comment]
Confimed fixed in https://github.com/OliveTin/OliveTin/pull/257 , and released in 2024.04.09.
